katdal/test/test_dataset.py

```diff
@@ -38,3 +94,48 @@ def test_selection_to_list():
     assert_equal(_selection_to_list(1), [1])
     # Groups
     assert_equal(_selection_to_list('all', all=['a', 'b']), ['a', 'b'])
+
+
+class TestVirtualSensors(object):
+    def setup(self):
+        self.target = Target('PKS1934-638, radec, 19:39, -63:42')
+        self.antennas = [Antenna('m000, -30:42:39.8, 21:26:38.0, 1086.6, 13.5, '
+                                 '-8.264 -207.29 8.5965'),
+                         Antenna('m063, -30:42:39.8, 21:26:38.0, 1086.6, 13.5, '
+                                 '-3419.5845 -1840.48 16.3825')]
+        corrprods = [('m000h', 'm000h'), ('m000v', 'm000v'),
+                     ('m063h', 'm063h'), ('m063v', 'm063v'),
+                     ('m000h', 'm063h'), ('m000v', 'm063v')]
+        subarray = Subarray(self.antennas, corrprods)
+        spw = SpectralWindow(centre_freq=1284e6, channel_width=0, num_chans=16,
+                             sideband=1, bandwidth=856e6)
+        self.timestamps = 1234667890.0 + 1.0 * np.arange(10)
+        self.dataset = MinimalDataSet(self.target, subarray, spw, self.timestamps)
+        self.array_ant = self.dataset.sensor.get('Antennas/array/antenna')[0]
+
+    def test_timestamps(self):
+        mjd = Timestamp(self.timestamps[0]).to_mjd()
+        assert_equal(self.dataset.mjd[0], mjd)
+        lst = self.array_ant.local_sidereal_time(self.timestamps)
+        assert_array_equal(self.dataset.lst, lst * (12 / np.pi))
+
+    def test_pointing(self):
+        az, el = self.target.azel(self.timestamps, self.antennas[1])
+        assert_array_equal(self.dataset.az[:, 1], rad2deg(az))
+        assert_array_equal(self.dataset.el[:, 1], rad2deg(el))
+        ra, dec = self.target.radec(self.timestamps, self.antennas[0])
+        assert_array_almost_equal(self.dataset.ra[:, 0], rad2deg(ra), decimal=5)
+        assert_array_almost_equal(self.dataset.dec[:, 0], rad2deg(dec), decimal=5)
+        angle = self.target.parallactic_angle(self.timestamps, self.antennas[0])
+        # TODO: Check why this is so poor...
+        assert_array_almost_equal(self.dataset.parangle[:, 0], rad2deg(angle), decimal=0)
+        x, y = self.target.sphere_to_plane(az, el, self.timestamps, self.antennas[1])
+        assert_array_equal(self.dataset.target_x[:, 1], rad2deg(x))
+        assert_array_equal(self.dataset.target_y[:, 1], rad2deg(y))
+
+    def test_uvw(self):
+        u, v, w = self.target.uvw(self.antennas[0], self.timestamps,
+                                  self.antennas[1])
+        assert_array_equal(self.dataset.u[:, 4], u)
+        assert_array_equal(self.dataset.v[:, 4], v)
+        assert_array_equal(self.dataset.w[:, 5], w)
```

Comment:
Why 4, 4, 5?

Reply:
Since corrprod 4 is `('m000h', 'm063h')` and corrprod 5 is `('m000v', 'm063v')`, they only differ in polarisation and should have identical `(u, v, w)`. This way I check both. Adding comment.